src/cpu/flexcpu/FlexCPU.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2018 The Regents of The University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# Authors: Bradley Wang
+
+
+from BaseCPU import BaseCPU
+from m5.params import *
+from m5.proxy import *
+
+class FlexCPU(BaseCPU):
+    type = 'FlexCPU'
+    cxx_header = 'cpu/flexcpu/flexcpu.hh'
+
+    clocked_cpu = Param.Bool(False, "Ties stages of the cpu to clock edges")
```

Comment:
Are these parameters ever used in our configurations? I know that you are thinking about how some one, some time, and some where, these parameters might be useful, but if we aren't actively using them or planning on using them in the near future, there's no need for this added code.

Reply:
It was for off-clock-edge behaviors when it was a dataflow design. I think we should actually default this to true, but I feel that we should keep it, since I think it's useful if we want to extend the reconfigurabiilty of the pipeline, since we could then choose what was latched or not.